### PR TITLE
linuxPackages.nvidia-x11: 495.46 -> 520.56.06

### PR DESCRIPTION
**Pull request by NickCao**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/nvidia-x11/default.nix [hunk 1]
```diff
@@ -27,11 +27,11 @@ rec {
   stable = if stdenv.hostPlatform.system == "i686-linux" then legacy_390 else latest;
 
   production = generic {
```
Comment (Kiskae) [inline pkgs/os-specific/linux/nvidia-x11/default.nix:29]: This is a "new feature branch" release, so you need to update the `latest` package, not `production`

**Review by Kiskae — CHANGES_REQUESTED** [review]

**Comment by Kiskae** [comment]
Release highlights since 515.76:

- Implemented support for over-the-air updates in the Proton and Wine NVIDIA NGX build. This feature is disabled by default and can be enabled by setting the “PROTON_ENABLE_NGX_UPDATER” environment variable to a value of “1”.
- Updated the Vulkan driver so that the following extensions no longer depend on nvidia-uvm.ko being loaded at runtime:
  - VK_KHR_acceleration_structure
  - VK_KHR_deferred_host_operations
  - VK_KHR_ray_query
  - VK_KHR_ray_tracing_pipeline
  - VK_NV_cuda_kernel_launch
  - VK_NV_ray_tracing
  - VK_NV_ray_tracing_motion_blur
  - VK_NVX_binary_import
  - VK_NVX_image_view_handle
- Updated nvidia-installer to allow use of the “–add-this-kernel” feature by non-root users.
- Updated nvidia-installer to display a more accurate progress bar when building the kernel modules.
- Updated nvidia-installer to display a warning message if a Vulkan ICD loader is not detected.
- Reworked nvidia-installer’s support for DKMS: the kernel modules will now be optionally registered with DKMS after the installer has already built and installed them on its own. nvidia-installer will now register the kernel modules with DKMS by default when the dkms(8) utility is detected on the system.
- Fixed a bug in the Vulkan driver which could lead to corruption in geometry and tessellation control shaders.
- Added a new CUDA Debugger implementation for Pascal and newer architectures as a part of the driver package: libcudadebugger.so (previously released separately as “CUDA GDB Developer Preview”).
- Fixed a regression in 515.76 that caused blank screens and hangs when starting an X server on RTX 30 series GPUs in some configurations where the boot display is connected via HDMI.
- Fixed a bug where Marvel’s Spider-Man Remastered would sometimes crash with Xid 13 errors on Turing and later.

Looks like `libcudadebugger.so` needs to be installed next to `libcuda.so`

**Comment by NickCao** [comment]
```
$ find /nix/store/yd8s4vjqjgmqf3lxfxh10ffl5x2f9kwk-nvidia-x11-520.56.06-6.0 -name 'libcuda*.so'
/nix/store/yd8s4vjqjgmqf3lxfxh10ffl5x2f9kwk-nvidia-x11-520.56.06-6.0/lib/libcuda.so
/nix/store/yd8s4vjqjgmqf3lxfxh10ffl5x2f9kwk-nvidia-x11-520.56.06-6.0/lib/libcudadebugger.so
```
looks like it's already there

**Comment by Kiskae** [comment]
Didn't realize the builder just installs all the `*.so` files in the distribution.
Although that might still not be enough because it looks like the soname is `libcudadebugger.so.1`, so applications that want to use it might not be able to locate it under that name.

**Comment by NickCao** [comment]
```
$ find /nix/store/yd8s4vjqjgmqf3lxfxh10ffl5x2f9kwk-nvidia-x11-520.56.06-6.0 -name 'libcuda*.so*'
/nix/store/yd8s4vjqjgmqf3lxfxh10ffl5x2f9kwk-nvidia-x11-520.56.06-6.0/lib/libcuda.so.520.56.06
/nix/store/yd8s4vjqjgmqf3lxfxh10ffl5x2f9kwk-nvidia-x11-520.56.06-6.0/lib/libcudadebugger.so.520.56.06
/nix/store/yd8s4vjqjgmqf3lxfxh10ffl5x2f9kwk-nvidia-x11-520.56.06-6.0/lib/libcuda.so
/nix/store/yd8s4vjqjgmqf3lxfxh10ffl5x2f9kwk-nvidia-x11-520.56.06-6.0/lib/libcuda.so.1
/nix/store/yd8s4vjqjgmqf3lxfxh10ffl5x2f9kwk-nvidia-x11-520.56.06-6.0/lib/libcudadebugger.so
/nix/store/yd8s4vjqjgmqf3lxfxh10ffl5x2f9kwk-nvidia-x11-520.56.06-6.0/lib/libcudadebugger.so.1
```
No worry, these links are also handled just fine.

**Comment by Kiskae** [comment]
Result of `nixpkgs-review pr 195778` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>bumblebee</li>
  </ul>
</details>
<details>
  <summary>100 packages built:</summary>
  <ul>
    <li>bottles</li>
    <li>geekbench</li>
    <li>geekbench4</li>
    <li>gl-gsync-demo</li>
    <li>gwe</li>
    <li>heroic</li>
    <li>linuxKernel.packages.linux_4_14.nvidia_x11</li>
    <li>linuxKernel.packages.linux_4_14.nvidia_x11_beta_open</li>
    <li>linuxKernel.packages.linux_4_14.nvidia_x11_production</li>
    <li>linuxKernel.packages.linux_4_14.nvidia_x11_production_open</li>
    <li>linuxKernel.packages.linux_4_14_hardened.nvidia_x11</li>
    <li>linuxKernel.packages.linux_4_14_hardened.nvidia_x11_beta_open</li>
    <li>linuxKernel.packages.linux_4_14_hardened.nvidia_x11_production</li>
    <li>linuxKernel.packages.linux_4_14_hardened.nvidia_x11_production_open</li>
    <li>linuxKernel.packages.linux_4_19.nvidia_x11</li>
    <li>linuxKernel.packages.linux_4_19.nvidia_x11_beta_open</li>
    <li>linuxKernel.packages.linux_4_19.nvidia_x11_production</li>
    <li>linuxKernel.packages.linux_4_19.nvidia_x11_production_open</li>
    <li>linuxKernel.packages.linux_4_19_hardened.nvidia_x11</li>
    <li>linuxKernel.packages.linux_4_19_hardened.nvidia_x11_beta_open</li>
    <li>linuxKernel.packages.linux_4_19_hardened.nvidia_x11_production</li>
    <li>linuxKernel.packages.linux_4_19_hardened.nvidia_x11_production_open</li>
    <li>linuxKernel.packages.linux_4_9.nvidia_x11</li>
    <li>linuxKernel.packages.linux_4_9.nvidia_x11_beta_open</li>
    <li>linuxKernel.packages.linux_4_9.nvidia_x11_production</li>
    <li>linuxKernel.packages.linux_4_9.nvidia_x11_production_open</li>
    <li>linuxKernel.packages.linux_5_10.nvidia_x11</li>
    <li>linuxKernel.packages.linux_5_10.nvidia_x11_beta_open</li>
    <li>linuxKernel.packages.linux_5_10.nvidia_x11_production</li>
    <li>linuxKernel.packages.linux_5_10.nvidia_x11_production_open</li>
    <li>linuxKernel.packages.linux_5_10_hardened.nvidia_x11</li>
    <li>linuxKernel.packages.linux_5_10_hardened.nvidia_x11_beta_open</li>
    <li>linuxKernel.packages.linux_5_10_hardened.nvidia_x11_production</li>
    <li>linuxKernel.packages.linux_5_10_hardened.nvidia_x11_production_open</li>
    <li>linuxKernel.packages.linux_5_15.nvidia_x11</li>
    <li>linuxKernel.packages.linux_5_15.nvidia_x11_beta_open</li>
    <li>linuxKernel.packages.linux_5_15.nvidia_x11_production</li>
    <li>linuxKernel.packages.linux_5_15.nvidia_x11_production_open</li>
    <li>linuxKernel.packages.linux_hardened.nvidia_x11 (linuxKernel.packages.linux_5_15_hardened.nvidia_x11)</li>
    <li>linuxKernel.packages.linux_hardened.nvidia_x11_beta_open (linuxKernel.packages.linux_5_15_hardened.nvidia_x11_beta_open)</li>
    <li>linuxKernel.packages.linux_hardened.nvidia_x11_production (linuxKernel.packages.linux_5_15_hardened.nvidia_x11_production)</li>
    <li>linuxKernel.packages.linux_hardened.nvidia_x11_production_open (linuxKernel.packages.linux_5_15_hardened.nvidia_x11_production_open)</li>
    <li>linuxKernel.packages.linux_5_19.nvidia_x11</li>
    <li>linuxKernel.packages.linux_5_19.nvidia_x11_beta_open</li>
    <li>linuxKernel.packages.linux_5_19.nvidia_x11_production</li>
    <li>linuxKernel.packages.linux_5_19.nvidia_x11_production_open</li>
    <li>linuxKernel.packages.linux_5_19_hardened.nvidia_x11</li>
    <li>linuxKernel.packages.linux_5_19_hardened.nvidia_x11_beta_open</li>
    <li>linuxKernel.packages.linux_5_19_hardened.nvidia_x11_production</li>
    <li>linuxKernel.packages.linux_5_19_hardened.nvidia_x11_production_open</li>
    <li>linuxKernel.packages.linux_5_4.nvidia_x11</li>
    <li>linuxKernel.packages.linux_5_4.nvidia_x11_beta_open</li>
    <li>linuxKernel.packages.linux_5_4.nvidia_x11_production</li>
    <li>linuxKernel.packages.linux_5_4.nvidia_x11_production_open</li>
    <li>linuxKernel.packages.linux_6_0.nvidia_x11</li>
    <li>linuxKernel.packages.linux_6_0.nvidia_x11_beta_open</li>
    <li>linuxKernel.packages.linux_6_0.nvidia_x11_production</li>
    <li>linuxKernel.packages.linux_6_0.nvidia_x11_production_open</li>
    <li>linuxKernel.packages.linux_latest_libre.nvidia_x11</li>
    <li>linuxKernel.packages.linux_latest_libre.nvidia_x11_beta_open</li>
    <li>linuxKernel.packages.linux_latest_libre.nvidia_x11_production</li>
    <li>linuxKernel.packages.linux_latest_libre.nvidia_x11_production_open</li>
    <li>linuxKernel.packages.linux_libre.nvidia_x11</li>
    <li>linuxKernel.packages.linux_libre.nvidia_x11_beta_open</li>
    <li>linuxKernel.packages.linux_libre.nvidia_x11_production</li>
    <li>linuxKernel.packages.linux_libre.nvidia_x11_production_open</li>
    <li>linuxKernel.packages.linux_lqx.nvidia_x11</li>
    <li>linuxKernel.packages.linux_lqx.nvidia_x11_beta_open</li>
    <li>linuxKernel.packages.linux_lqx.nvidia_x11_production</li>
    <li>linuxKernel.packages.linux_lqx.nvidia_x11_production_open</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.nvidia_x11</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.nvidia_x11_beta_open</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.nvidia_x11_production</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.nvidia_x11_production_open</li>
    <li>linuxKernel.packages.linux_xanmod.nvidia_x11</li>
    <li>linuxKernel.packages.linux_xanmod.nvidia_x11_beta_open</li>
    <li>linuxKernel.packages.linux_xanmod.nvidia_x11_production</li>
    <li>linuxKernel.packages.linux_xanmod.nvidia_x11_production_open</li>
    <li>linuxKernel.packages.linux_xanmod_latest.nvidia_x11</li>
    <li>linuxKernel.packages.linux_xanmod_latest.nvidia_x11_beta_open</li>
    <li>linuxKernel.packages.linux_xanmod_latest.nvidia_x11_production</li>
    <li>linuxKernel.packages.linux_xanmod_latest.nvidia_x11_production_open</li>
    <li>linuxKernel.packages.linux_xanmod_stable.nvidia_x11</li>
    <li>linuxKernel.packages.linux_xanmod_stable.nvidia_x11_beta_open</li>
    <li>linuxKernel.packages.linux_xanmod_stable.nvidia_x11_production</li>
    <li>linuxKernel.packages.linux_xanmod_stable.nvidia_x11_production_open</li>
    <li>linuxKernel.packages.linux_xanmod_tt.nvidia_x11</li>
    <li>linuxKernel.packages.linux_xanmod_tt.nvidia_x11_beta_open</li>
    <li>linuxKernel.packages.linux_xanmod_tt.nvidia_x11_production</li>
    <li>linuxKernel.packages.linux_xanmod_tt.nvidia_x11_production_open</li>
    <li>linuxKernel.packages.linux_zen.nvidia_x11</li>
    <li>linuxKernel.packages.linux_zen.nvidia_x11_beta_open</li>
    <li>linuxKernel.packages.linux_zen.nvidia_x11_production</li>
    <li>linuxKernel.packages.linux_zen.nvidia_x11_production_open</li>
    <li>mangohud</li>
    <li>monitor</li>
    <li>nvfancontrol</li>
    <li>primus</li>
    <li>psensor</li>
    <li>xpraWithNvenc</li>
  </ul>
</details>

**Review by Kiskae — APPROVED** [review]

File: pkgs/os-specific/linux/nvidia-x11/default.nix [hunk 2]
```diff
@@ -37,10 +37,11 @@ rec {
   };
 
   latest = selectHighestVersion production (generic {
-    version = "495.46";
```
Comment (PedroHLC) [inline pkgs/os-specific/linux/nvidia-x11/default.nix:40]: The commit diff does not match the PR title :eyes:

**Review by PedroHLC — APPROVED** [review]

**Pull request merged**